configure.ac

```diff
@@ -1,313 +1,127 @@
-# Process this file with autoconf to produce a configure script.
+dnl Process this file with autoconf to produce a configure script.
 
 AC_PREREQ(2.59)
+
 AC_INIT(Normaliz, 3.7.4)
+
 AC_SUBST(LIBNORMALIZ_VERSION_MAJOR, 3)
 AC_SUBST(LIBNORMALIZ_VERSION_MINOR, 7)
 AC_SUBST(LIBNORMALIZ_VERSION_PATCH, 4)
 AC_SUBST(LIBNORMALIZ_VERSION_STRING, "$PACKAGE_VERSION")
-AM_INIT_AUTOMAKE(foreign)
-AC_CONFIG_FILES([Makefile
-	source/Makefile
-	source/libnormaliz/version.h
-	example/Makefile
-	test/Makefile
-	source/libnormaliz/nmz_config.h])
+
 AC_CONFIG_MACRO_DIR([m4])
+AC_CONFIG_AUX_DIR([.])
+AM_INIT_AUTOMAKE([1.14.1])
+
+AC_CANONICAL_HOST
+
+dnl Find C++ Compiler
 AC_PROG_CXX
+
+dnl Set INSTALL or use install-sh.
+AC_PROG_INSTALL
+
+dnl We build our library with libtool.
+LT_INIT
+
+dnl We want to use some C++14 features
 AX_CXX_COMPILE_STDCXX(14, , optional)
+
 AS_IF([test x$HAVE_CXX14 = x0],
+  dnl We require C++0x features
   [ AX_CXX_COMPILE_STDCXX(0x, , mandatory) ])
 
-AC_PROG_LIBTOOL
+dnl Check for required/optional libraries
 AC_LANG(C++)
 
-## Test for GMP
-AC_ARG_WITH(gmp,
-   AS_HELP_STRING([--with-gmp=DIR],
-                  [Use the GMP library installed in installation prefix DIR.]),
-   [ LDFLAGS="-L$with_gmp/lib $LDFLAGS"
-     CPPFLAGS="-I $with_gmp/include $CPPFLAGS"
-   ])
-AC_MSG_CHECKING([whether GMP C++ library is available])
-LIBS_SAVED="$LIBS"
-LIBS="$LIBS -lgmpxx -lgmp"
-AC_LINK_IFELSE(
-   [AC_LANG_PROGRAM([[#include <gmpxx.h>
-                     ]],
-                    [[mpz_class a;
-                     ]])],
-   [AC_MSG_RESULT([yes])],
-   [AC_MSG_ERROR([GMP C++ library not found. Make sure it was compiled with --enable-cxx])])
-LIBS="$LIBS_SAVED"
+AC_CHECK_HEADERS([boost/dynamic_bitset.hpp], , AC_MSG_ERROR([boost headers not found]))
 
-## Test for OpenMP parallelization.
-AC_ARG_ENABLE([openmp],
-    [AS_HELP_STRING([--enable-openmp@<:@=ARG@:>@],
-      [enable parallelization with OpenMP @<:@default=check@:>@])],
-    [:],
-    [enable_openmp=check])
-AS_IF([test "x$enable_openmp" != xno],
-  [AX_OPENMP([have_openmp=yes],
-	     [have_openmp=no])])
+# We use GMP's C++ wrapper but we also interface with GMP's C interface directly.
+AC_CHECK_LIB([gmp], [__gmpz_init], [have_gmp=yes], AC_MSG_ERROR([GMP library not found]))
 
-AS_IF([test x$have_openmp = xyes],
-  [enable_openmp=yes],
-  [AS_IF([test "x$enable_openmp" = xyes],
-    [AC_MSG_ERROR([openmp not available])],
-    [AS_IF([test "x$enable_openmp" != xno],
-        [AC_MSG_NOTICE([Parallelization with OpenMP will not be available because no suitable compiler was found])])
-     enable_openmp=no])])
+# GMPXX does not contain anything that we can check for with AX_CXX_CHECK_LIB
+# so we just check for something from the standard library, i.e., that -lgmpxx
+# goes through
+AX_CXX_CHECK_LIB([gmpxx], [std::runtime_error::what () const], [have_gmpxx=yes], AC_MSG_ERROR([GMP library for C++ not found]), [-lgmp])
```

Comment:
Could you elaborate on that? Why can't you check for `mpz_class`?

Reply:
`mpz_class` is not a symbol, it's a typedef that resolves to something very complicated.

Reply:
Relevant should be a symbol from libgmpxx which is needed by Normaliz. So I tried linking without it, and the only missing symbol it complained about was `operator>>(std::__1::basic_istream<char, std::__1::char_traits<char> >&, __mpq_struct*)`. So that's what I'd test for. If `AX_CXX_CHECK_LIB` can't do that, I wouldn't use `AX_CXX_CHECK_LIB`, and just roll the code for it myself (something like `mpq_class tmp; std::cin >> tmp;` should probably do the trick).

Reply:
I think that's complicating things too much as we're not searching for all the used symbols elsewhere. I don't think you'll find a system that has a working `libgmp` but a `libgmpxx` that is missing that symbols. But feel free to add such a check if you think it's a good idea.